aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -571,6 +571,100 @@ Tensor& cholesky_out(Tensor &result, const Tensor &self, bool upper) {
   return result;
 }
 
+template<typename scalar_t>
+static void  apply_cholesky_mod(const Tensor& self, Tensor & out, Tensor & err, bool upper, std::vector<int64_t>& infos) {
+#ifndef USE_LAPACK
+  AT_ERROR("cholesky: LAPACK library not found in compilation");
+#else
+  char uplo = upper ? 'U' : 'L';
+
+  Tensor outTag =  out;
+  Tensor selfTag = self;
+
+  double* err_data = err.data_ptr<double>();
+  auto self_matrix_stride = matrixStride(self);
+  auto batch_size = batchCount(self);
+  auto n = self.size(-2);
+  auto m = self.size(-1);
+
+  if (batch_size > 1) {
+    outTag =  out.flatten(0, -3);
+    selfTag = self.flatten(0, -3);
+  }
+  
+  scalar_t* self_data = selfTag.data_ptr<scalar_t>();
+  scalar_t* out_data = outTag.data_ptr<scalar_t>();
+  /* TODO: do this in parallel */
```

Comment:
this seems like an important TODO - does the current vanilla cholesky implementation do this in parallel?

Reply:
No, the vanilla Cholesky doesn't do this in parallel.

Reply:
No need to do it in parallel so that all the threads are utilized in the kernels, which usually dominate the overall computation.